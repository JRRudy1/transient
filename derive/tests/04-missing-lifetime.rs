

//! Tests the behavior when used on structs without a lifetime parameter
use transient_any_derive::MakeStatic;


#[derive(Debug, Clone, PartialEq, Eq, MakeStatic)]
struct S<T> {
    value: T,
}

type SS = S<String>;


fn main() {
    // this test should fail to compile
}
