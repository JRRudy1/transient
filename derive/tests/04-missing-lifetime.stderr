error: Exactly one lifetime parameter is required!
 --> tests/04-missing-lifetime.rs:8:9
  |
8 | struct S<T> {
  |         ^
